In [1]:
import pandas as pd
import numpy as np


Reading the data:

In [3]:
df = pd.read_csv("E:\Guvi\Projects\Inventory forecasting\\train 2.csv")
df

date  store  item  sales
0       2013-01-01      1     1     13
1       2013-01-02      1     1     11
2       2013-01-03      1     1     14
3       2013-01-04      1     1     13
4       2013-01-05      1     1     10
...            ...    ...   ...    ...
912995  2017-12-27     10    50     63
912996  2017-12-28     10    50     59
912997  2017-12-29     10    50     74
912998  2017-12-30     10    50     62
912999  2017-12-31     10    50     82

[913000 rows x 4 columns]

Exploratory Data Analysis(EDA)

In [4]:
df.nunique()

date     1826
store      10
item       50
sales     213
dtype: int64

findings: 
We can see that the data is clean of outliers and duplicates. Since, there will be 
1. 1826 days in 5 years period 
2. As per the data we have 10 unique stores and 50 unique products. 
3. We can not tell whether all the values in sales are right/wrong.

In [9]:
df.dtypes

date     object
store     int64
item      int64
sales     int64
dtype: object

Finding: We can see that all the columns are in 'int64' datatype other than date which is in 'object' datatype.

In [11]:
df['date'] = pd.to_datetime(df['date'])
df.dtypes

date     datetime64[ns]
store             int64
item              int64
sales             int64
dtype: object

Changed date column to 'datetime' format for better interpretation. 

In [15]:
df['cumsum_sales'] = df['sales'].cumsum()
df

date  store  item  sales  cumsum_sales
0      2013-01-01      1     1     13            13
1      2013-01-02      1     1     11            24
2      2013-01-03      1     1     14            38
3      2013-01-04      1     1     13            51
4      2013-01-05      1     1     10            61
...           ...    ...   ...    ...           ...
912995 2017-12-27     10    50     63      47704235
912996 2017-12-28     10    50     59      47704294
912997 2017-12-29     10    50     74      47704368
912998 2017-12-30     10    50     62      47704430
912999 2017-12-31     10    50     82      47704512

[913000 rows x 5 columns]